### DISCUSSION TITLE
How to implement multiple selection in submenu?

### DISCUSSION THREAD

**Discussion by KrabsWong** [microsoft/vscode-discussions#140] [Extension Development QnA] [answered]
Like  the screenshot below, I've added a submenu normally.

What I want to do next is do a filter after use check the submenu item, is there a way to support multiple select for the submenu? like the sidebar view selection as the second screenshot below.

<img width="748" alt="image" src="https://user-images.githubusercontent.com/12368943/192097456-c7c2e88d-01d6-4b95-9be8-4db61684b492.png">

<img width="690" alt="image" src="https://user-images.githubusercontent.com/12368943/192097514-55f61015-d0aa-45d8-9d2c-5913ea4f87c5.png">

**Comment by Tommycore** [comment]
I haven't worked with submenus, but when you say "treeview in panel", are we talking about a regular `vscode.TreeDataProvider`? Because if so, the solution is easy:
```typescript
vscode.window.createTreeView('smithy.schemaView', {
  treeDataProvider: myTreeDataProviderInstance,
  canSelectMany: true, // <- This is the important part.
  showCollapseAll: true,
});
```

**Reply by KrabsWong** [reply]
Thanks bro.

But it doesn't match my requirement, I wanna the menu itself can support multiple selection. Anyway, many thansk.

**Comment by gjsjohnmurray** [comment, marked as answer]
See [#109306](https://github.com/microsoft/vscode/issues/109306) for an open issue that asks for this to be made possible.
